lending/loan_management/doctype/loan_repayment/loan_repayment.py

```diff
@@ -2984,6 +2980,48 @@ def get_accrued_interest(
 	return flt(accrued_interest)
 
 
+def get_unbooked_interest_for_bulk_loans(loans, posting_date):
+
+	loan_demand_doc = frappe.qb.DocType("Loan Demand")
+	query1 = (
+		frappe.qb.from_(loan_demand_doc)
+		.where(loan_demand_doc.docstatus == 1)
+		.where(loan_demand_doc.demand_subtype == "Interest")
+		.where(loan_demand_doc.demand_date <= posting_date)
+		.where(loan_demand_doc.loan.isin(loans))
+		.orderby(loan_demand_doc.demand_date, order=Order.desc)
+		.select(
+			loan_demand_doc.loan.as_("loan"), Max(loan_demand_doc.demand_date).as_("last_demand_date")
+		)
+	)
+
+	query2 = (
```

Comment:
Use better variable names. Every query is a query, this just be some specific thing.